app/assets/javascripts/controllers/project_roles_ctrl.js

```diff
@@ -0,0 +1,126 @@
+samson.controller('ProjectRolesCtrl', function ($scope, $element, $filter, projectRolesService, messageCenterService) {
+    $scope.project_role = {};
+    $scope.roles = [];
+
+    $scope.initModel = function () {
+        $scope.project_role = {
+            id: getUniqueId(),
+            user_id: getUserId(),
+            project_id: getProjectId(),
+            role_id: getRoleId()
+        };
+
+        loadProjectRoles();
+    };
+
+    $scope.$watch('project_role.role_id', function (new_role_value, old_role_value) {
+        if (new_role_value !== old_role_value) {
```

Comment:
This code seems a bit strange. The function will only return a value on this branch? What will happen if the `new_role_value === old_role_value`? It should return something no?